.github/workflows/submit.yml

```diff
@@ -602,346 +478,70 @@ jobs:
-          mkdir -p "${HOME}/bootjdk/${BOOT_JDK_VERSION}"
-          wget -O "${HOME}/bootjdk/${BOOT_JDK_FILENAME}" "${BOOT_JDK_URL}"
-          echo "${BOOT_JDK_SHA256} ${HOME}/bootjdk/${BOOT_JDK_FILENAME}" | sha256sum -c >/dev/null -
-          tar -xf "${HOME}/bootjdk/${BOOT_JDK_FILENAME}" -C "${HOME}/bootjdk/${BOOT_JDK_VERSION}"
-          mv "${HOME}/bootjdk/${BOOT_JDK_VERSION}/"*/* "${HOME}/bootjdk/${BOOT_JDK_VERSION}/"
-        if: steps.bootjdk.outputs.cache-hit != 'true'
-
-      - name: Restore build JDK
-        id: build_restore
-        uses: actions/download-artifact@v2
-        with:
-          name: transient_jdk-linux-x64_${{ needs.prerequisites.outputs.bundle_id }}
-          path: ~/jdk-linux-x64
-        continue-on-error: true
-
-      - name: Unpack jdk
-        run: |
-          mkdir -p "${HOME}/jdk-linux-x64/jdk-${{ env.JDK_VERSION }}-internal+0_linux-x64_bin"
-          tar -xf "${HOME}/jdk-linux-x64/jdk-${{ env.JDK_VERSION }}-internal+0_linux-x64_bin.tar.gz" -C "${HOME}/jdk-linux-x64/jdk-${{ env.JDK_VERSION }}-internal+0_linux-x64_bin"
-
-      - name: Find root of build JDK image dir
-        run: |
-          build_jdk_root=`find ${HOME}/jdk-linux-x64/jdk-${{ env.JDK_VERSION }}-internal+0_linux-x64_bin -name release -type f`
-          echo "build_jdk_root=`dirname ${build_jdk_root}`" >> $GITHUB_ENV
+          sudo apt-get install gcc-10=10.2.0-5ubuntu1~20.04 g++-10=10.2.0-5ubuntu1~20.04 libxrandr-dev libxtst-dev libcups2-dev libasound2-dev
+          sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 100 --slave /usr/bin/g++ g++ /usr/bin/g++-10
+        if: matrix.debian-arch == ''
 
       - name: Install cross-compilation host dependencies
-        run: |
-          sudo apt-get update
-          sudo apt-get install g++-s390x-linux-gnu gcc-s390x-linux-gnu
+        run: sudo apt-get install gcc-10-${{ matrix.gnu-arch}}-linux-gnu${{ matrix.gnu-flavor}}=10.2.0-5ubuntu1~20.04cross1 g++-10-${{ matrix.gnu-arch}}-linux-gnu${{ matrix.gnu-flavor}}=10.2.0-5ubuntu1~20.04cross1
+        if: matrix.debian-arch != ''
 
       - name: Cache sysroot
         id: cache-sysroot
         uses: actions/cache@v2
         with:
-          path: ~/sysroot-s390x/
-          key: sysroot-s390x-${{ hashFiles('jdk/.github/workflows/submit.yml') }}
+          path: ~/sysroot-${{ matrix.debian-arch}}/
+          key: sysroot-${{ matrix.debian-arch}}-${{ hashFiles('jdk/.github/workflows/submit.yml') }}
+        if: matrix.debian-arch != ''
 
       - name: Install sysroot host dependencies
-        run: |
-          sudo apt-get install debootstrap qemu-user-static
-        if: steps.cache-sysroot.outputs.cache-hit != 'true'
+        run: sudo apt-get install debootstrap qemu-user-static
+        if: matrix.debian-arch != '' && steps.cache-sysroot.outputs.cache-hit != 'true'
 
       - name: Create sysroot
         run: >
           sudo qemu-debootstrap
-          --arch=s390x
+          --arch=${{ matrix.debian-arch}}
```

Comment:
Style: `${{ matrix.debian-arch }}` (space before closing braces) -- also in other places.

Reply:
Ah right, good catch, fixed these.